code/modules/food_and_drinks/machinery/microwave.dm

```diff
@@ -466,20 +477,36 @@
 	. = ..()
 	if((machine_stat & NOPOWER) && operating)
 		pre_fail()
-		eject()
+		eject(force = TRUE) // monkestation edit: microwave "enhancements"
 
 /obj/machinery/microwave/proc/loop_finish(mob/cooker)
 	operating = FALSE
 
 	var/metal_amount = 0
+	var/shouldnt_open = FALSE // monkestation edit: microwave "enhancements"
+	var/dont_eject = FALSE // monkestation edit: microwave "enhancements"
 	for(var/obj/item/cooked_item in ingredients)
+		// var/sigreturn = cooked_item.microwave_act(src, cooker, randomize_pixel_offset = ingredients.len)
+		// if(sigreturn & COMPONENT_MICROWAVE_SUCCESS)
+		// 	if(isstack(cooked_item))
+		// 		var/obj/item/stack/cooked_stack = cooked_item
+		// 		dirty += cooked_stack.amount
+		// 	else
+		// 		dirty++
+		// monkestation edit start: microwave "enhancements"
 		var/sigreturn = cooked_item.microwave_act(src, cooker, randomize_pixel_offset = ingredients.len)
 		if(sigreturn & COMPONENT_MICROWAVE_SUCCESS)
+			var/should_dirty = !(sigreturn & COMPONENT_MICROWAVE_DONTDIRTY)
 			if(isstack(cooked_item))
 				var/obj/item/stack/cooked_stack = cooked_item
-				dirty += cooked_stack.amount
+				should_dirty && (dirty += cooked_stack.amount)
 			else
-				dirty++
+				should_dirty && (dirty++)
```

Comment:
I get what this is doing (lord knows if it actually does) but you shouldn't be using an && when this is what if statements exist for